File: framework/languages/java/e2e/SpotService/Client/src/main/java/systems/zlink/e2e/spotservice/client/Scenarios/SmA2Scenario.java
package systems.zlink.e2e.spotservice.client.Scenarios;

import systems.zlink.e2e.spotservice.shared.Contracts;

public final class SmA2Scenario extends SpotServiceScenarioContext {
    private SmA2Scenario(SpotServiceScenarioContext context) {
        super(context);
    }

    public static void run(SpotServiceScenarioContext context) {
        new SmA2Scenario(context).execute();
    }

    private void execute() {
        Contracts.StateRes second = eventually(() -> requestState("room-a", "a2", REQUEST_TIMEOUT));
        ensure(second.value().contains("a1") && second.value().contains("a2"),
            "SM-A2 state did not accumulate");
        System.out.println("scenario SM-A2 passed");

    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Client/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/client/scenarios/SmA2Scenario.kt
package systems.zlink.e2e.kotlin.spotservice.client.scenarios

import systems.zlink.e2e.kotlin.spotservice.client.support.ensure
import systems.zlink.e2e.kotlin.spotservice.client.support.eventually
import systems.zlink.e2e.kotlin.spotservice.client.support.SpotHttpDriver

internal object SmA2Scenario {
    suspend fun run(spots: SpotHttpDriver) {
        val second = eventually {
            spots.requestState("room-a", "a2")
        }
        ensure(
            second.value.contains("a1") && second.value.contains("a2"),
            "SM-A2 state did not accumulate",
        )
        println("scenario SM-A2 passed")
    }
}
